src/Processor/ProgressiveProcessor.php

```diff
@@ -19,12 +19,7 @@ public function process(Image $image, Request $request): void
                                         $request->query->get('progressive') :
                                         $request->query->get('interlace');
 
-            /*
-             * Upgrade Intervention Image to 3.x
-             * interlace() no longer exists and is handle by encoder options.
-             * @see https://image.intervention.io/v3/basics/image-output
-             */
-            $image->interlace((bool) $process);
+            $image->encodeByMediaType(progressive: (bool) $process);
```

Comment:
I think this logic must be moved from Processors to an Event Subscriber when Image is rendered or saved.